### ISSUE TITLE
Logloss constant predictions baseline

### ISSUE THREAD

**Issue by cusinato** [facebookresearch/PyTorch-BigGraph#158]
I'm monitoring the logloss and eval stats during PBG training with num_batch_negs=50 and num_uniform_negs=50. Do I need to consider the class imbalance of the training batches passed to the logistic classifier to better understand the meaning of the logloss values? My goal here is to know if the logloss PBG logistic model logloss is at least better than a intercept-only logistic classifier (constant predictions) baseline logloss like described in https://stats.stackexchange.com/questions/276067/whats-considered-a-good-log-loss

Now I'm considering the balanced logloss baseline (0,69) because: "The total loss of due to the negatives is renormalized so it compares with the one of the positives" https://torchbiggraph.readthedocs.io/en/latest/loss_optimization.html#logistic-loss

So in this setting with 100 negatives edges (50 batch + 50 uniform) per positive edge, do I need to consider a class prevalence of 1% and compare my logloss with a intercept-only logloss baseline of 0,06 or am I right to consider a prevalence of 50% and the intercept-only logloss baseline of 0,69 due to this renormalized loss?

My config file:

```
def get_torchbiggraph_config():
    config = dict(
        # I/O data
        entity_path="pbghelper/output/metadata", 
        edge_paths=[
            "pbghelper/output/edges", 
        ],
        checkpoint_path="pbg/mod20201014/checkpoint",
        checkpoint_preservation_interval=1,

        # Graph structure
        entities={
            'all': {'num_partitions': 9},
        },
        relations=[{'name': str(uu), 'lhs':'all', 'rhs': 'all', 'operator': 'translation'} for uu in [_ for _ in range(9)]],
        dynamic_relations=False,

        # Scoring model
        dimension=39,
        global_emb=False,
        comparator='dot',
        init_scale=0.1,
        bias=True,

        # Training
        num_epochs=33,
        workers=60,
        batch_size=500,
        num_batch_negs=50,
        num_uniform_negs=50,
        max_edges_per_chunk=250000000,
        loss_fn='logistic',
        lr=0.015,
        eval_fraction=0.007,
    )

    return config
```

**Comment by adamlerer** [comment]
You can read the code here
https://github.com/facebookresearch/PyTorch-BigGraph/blob/master/torchbiggraph/losses.py#L51

I believe the log loss will be

-log( Q(pos) ) - 1/100 * sum_{i=1}^{100} { log( 1 - Q(neg_i) ) } . For a constant classifier Q, that's minimized when Q(.) = 0.5 , and has a log loss of 2 log(0.5) = 1.39 .

If you want to check this, you can randomly shuffle the labels (robust) or set lr=0 (easy), you should get a loss around that value.
